### PR TITLE
Use queue to replace the recursion call in sync progress

### PR DESCRIPTION
**Pull request by jackzhhuang**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: types/src/block/mod.rs [hunk 1]
```diff
@@ -402,21 +397,25 @@ impl BlockHeader {
     }
 
     //for test
```
Comment (nkysg) [inline types/src/block/mod.rs:399]: why not use [cfg(test)]

File: types/src/block/mod.rs [hunk 2]
```diff
@@ -402,21 +397,25 @@ impl BlockHeader {
...
-    pub fn dag_genesis_random() -> Self {
```
Comment (nkysg) [inline types/src/block/mod.rs:405]: why not use [cfg(test)]

File: types/src/block/mod.rs [hunk 3]
```diff
@@ -402,21 +397,25 @@ impl BlockHeader {
...
+    pub fn dag_genesis_random(dag_genesis_number: BlockNumber) -> Self {
         let mut header = Self::random();
         header.parents_hash = None;
-        header.number = get_test_flexidag_fork_height();
+        header.number = dag_genesis_number;
         header
     }
 
     //for test
-    pub fn dag_genesis_random_with_parent(parent: BlockHeader) -> Self {
```
Comment (nkysg) [inline types/src/block/mod.rs:413]: the same with previous

File: types/src/block/mod.rs (hunk 2 above)
Comment (jackzhhuang) [reply types/src/block/mod.rs:405]: [cfg(test)] cannot be used by other modules.

Comment (nkysg) [reply types/src/block/mod.rs:405]: got it. But I think function dag_genesis_random and dag_genesis_random_with_parent only use in flexidag/tests/test.rs, can we move it to releated files. Maybe it does't need to make public to other mods.

Comment (jackzhhuang) [reply types/src/block/mod.rs:405]: Ok, I will try.

**Review by sanlee42 — APPROVED** [review]

**Pull request merged**